cpp/src/arrow/compute/kernels/CMakeLists.txt

```diff
@@ -72,12 +72,20 @@ add_arrow_compute_test(vector_test
                        vector_hash_test.cc
                        vector_nested_test.cc
                        vector_replace_test.cc
-                       vector_selection_test.cc
-                       vector_sort_test.cc
                        vector_run_end_encode_test.cc
                        select_k_test.cc
                        test_util.cc)
 
+add_arrow_compute_test(vector_sort_test
+                       SOURCES
+                       vector_sort_test.cc
+                       test_util.cc)
+
+add_arrow_compute_test(vector_selection_test
+                       SOURCES
+                       vector_selection_test.cc
+                       test_util.cc)
```

Comment:
At some point we might want to create an object library for `test_util.cc` instead of recompiling it for each target. Can be left as a subsequent PR.

Reply:
That's possible. We can revisit later :-)